### ISSUE TITLE
lerna version - "error: unable to sign the tag"

### ISSUE THREAD

**Issue by ernestostifano** [lerna/lerna#2554]
Lerna fails to sign tags when running `lerna version`.

## Expected Behavior

Tags to be correctly signed as well as commits (commits are successfully signed).

## Current Behavior

```
lerna ERR! Error: Command failed: git tag @scope/my-package@1.0.0-beta.1 -m @scope/my-package@1.0.0-beta.1
lerna ERR! error: gpg failed to sign the data
lerna ERR! error: unable to sign the tag
```
Please note that the same command (`git tag ...`) works fine and succesfully sign tags when manually ran.

## Possible Solution

Some ideas...
* Is `lerna version` running tasks in child processes? Does those processes have access to the GPG agent (GPG Suite in this case)?
* Commit signing seems to be correctly working, that should be a good staring point.
* What user information (name/email) is being used? (GPG key to be used is matched using that info).

## Steps to Reproduce (for bugs)

1. Correctly setup GPG and set commit.gpgSign and tag.gpgSign to true in global git config.
2. Run `lerna version` on project.
3. Follow steps. Process will fail when trying to sign tags.

<details><summary>lerna.json</summary><p>

```json
{
    "packages": ["./packages/*"],
    "version": "independent",
    "useWorkspaces": true,
    "npmClient": "yarn"
}
```
</p></details>

## Your Environment

| Executable | Version |
| ---: | :--- |
| `lerna --version` | 3.20.2 |
| `npm --version`  | 6.14.4 |
| `yarn --version` | 1.22.4 |
| `node --version` | 13.12.0 |

| OS | Version |
| --- | --- |
| NAME | VERSION |
| macOS Catalina | 10.15.3 |

**Comment by carsonfarmer** [comment]
I'm seeing this exact same behavior with `npm` (as opposed to `yarn`).

**Comment by ryexley** [comment]
I'm running into the same issue, also using Yarn. Commits are signed with no issues, but tagging fails. We would really prefer to enforce having our tags signed. Has anyone had any success getting tag signing to work without failing?

**Comment by skyrpex** [comment]
Same problem here: the tagging fails when ran using lerna, but works correctly when manually ran.

**Comment by skyrpex** [comment]
Strangely enough, I placed a console log at `@lerna/version/lib/git-tag.js` because I wanted to manually reproduce the problem, so it looks like this:

```
  log.verbose("git", args);
  console.log("git", args, opts);
  return childProcess.exec("git", args, opts);
```

And it worked now! It successfully tagged my 14 repos.

Why would the console log make it work? :thinking:

**Comment by ryexley** [comment]
@skyrpex that, to me, makes it sound like it could potentially be a timing issue, a race condition. hhmmm

**Comment by skyrpex** [comment]
@ryexley yeah, that's what I'm thinking. Didn't try to just add a small delay, but I bet it would work...

**Comment by rs-8** [comment]
same thing

**Comment by byhow** [comment]
I am getting the same issue as well. Any updated resolution or possible causes?

**Comment by byhow** [comment]
After cleaning up all the GPG keys I have from `~/.gnupg/private-keys-v1.d/` and regenerating a new pair, it seems to get rid of the issue. It doesn't seem to be a race issue for me but probably a signing key setup issue, so ymmv
